### DISCUSSION TITLE
Open commit suggestions are put into "hidden conversations" while resolved are presented

### DISCUSSION THREAD

**Discussion by matkoniecz** [community/community#11842] [Pull Requests]
Open https://github.com/streetcomplete/StreetComplete/pull/3648

Note 

![screen01](https://user-images.githubusercontent.com/899988/154910062-3aa45c99-5f97-4cbc-991b-5ffdbc2b66b7.png)

Note log list of recent resolved commit suggestions.

Use "Load more..."

Note that some hidden commit suggestions are open.

Commit suggestions not marked as resolved should be shown in preference to ones marked as resolved, even if older.

**Comment by danfishgold** [comment]
this is a really annoying problem. my team members have already missed several review comments because they were hidden

even when I send them a link to a specific comment that they missed (something like org/repo/pull/123#discussion_r12345678), the link doesn't auto-expand hidden comments so I can't really work around this problem

**Reply by davidmigloz** [reply]
I see this happening again and again, colleagues missing unresolved conversations in PRs because they are hidden. 
If a conversation is unresolved, what's the purpose of hidden it?

**Comment by matkoniecz** [comment]
https://github.com/streetcomplete/StreetComplete/pull/4642 - some quite large PR with many reviews, GitHub again gets confused (117 comments, 250 commits proposed to be added)

![Peek 2023-02-14 06-38](https://user-images.githubusercontent.com/899988/218649064-62fd9323-e94b-4cd5-9fb5-6eb0d9cffc48.gif)

even linking to specific discussion like https://github.com/streetcomplete/StreetComplete/pull/4642#discussion_r1036289502 https://github.com/streetcomplete/StreetComplete/pull/4642#discussion_r1056497728 https://github.com/streetcomplete/StreetComplete/pull/4642#discussion_r1056671016 https://github.com/streetcomplete/StreetComplete/pull/4642#discussion_r1057245664 sometimes fails (no pattern found so far)

It seems that seemingly "Outdated" are especially prone to disappearing even when discussion itself is relevant (and that is why it was NOT marked as resolved!)

But not only them! As bonus, wait time for showing hidden comments is several seconds after you request unrolling comments...

![Peek 2023-02-14 08-38](https://user-images.githubusercontent.com/899988/218670509-1d11cf5d-db14-488b-8d51-164cfe31a287.gif)

**Comment by odahcam** [comment]
I think comments should always be shown next to the lines they were made even if new commits are added. There could be a reference old line or something but the comments should always show unless resolved, otherwise there's no value in commenting a PR if my comment is gonna vanish after a new commit is added, I'll have to review everything anyways.

**Comment by ChrisMacGregorONI** [comment]
https://github.com/orgs/community/discussions/130618#discussioncomment-12105463
